File: bruce/p.cpp
#include <iostream>
#include <cmath>
#include <array>

using namespace std;

static array<double, 3> to_cart(double lat, double lon)
{
    lat *= M_PI / 180.0;
    lon *= M_PI / 180.0;
    return {{
        cos(lat) * cos(lon), cos(lat) * sin(lon), sin(lat)
    }};
}

static double dist(const array<double, 3> &a, const array<double, 3> &b)
{
    double d = a[0] * b[0] + a[1] * b[1] + a[2] * b[2];
    return acos(d);
}

static double dist(double lat1, double lon1, double lat2, double lon2)
{
    return dist(to_cart(lat1, lon1), to_cart(lat2, lon2));
}

int main()
{
    int N;
    double lat, lon, lat0, lon0;
    double ans = 0;
    cin >> N;
    cin >> lat >> lon;
    lat0 = lat; lon0 = lon;
    for (int i = 1; i < N; i++)
    {
        double lat2, lon2;
        cin >> lat2 >> lon2;
        ans += dist(lat, lon, lat2, lon2);
        lat = lat2;
        lon = lon2;
    }
    ans += dist(lat, lon, lat0, lon0);
    ans *= 6371.0;
    cout << fixed << ans << '\n';
}

File: p.py
# p.py 
# Nicholas Pilkington

from math import radians, cos, sin, asin, sqrt
import os, sys

def haversine(lon1, lat1, lon2, lat2):

    lon1, lat1, lon2, lat2 = map(radians, [lon1, lat1, lon2, lat2])

    dlon = lon2 - lon1 
    dlat = lat2 - lat1 
    a = sin(dlat/2)**2 + cos(lat1) * cos(lat2) * sin(dlon/2)**2
    c = 2.0 * asin(sqrt(a)) 

    # 6371 km is the radius of the Earth
    km = 6371 * c
    return km 

N = None
coords = []
solution = 0.0

with open(sys.argv[1], mode='r') as fd:
    N = int(fd.readline())

    for i in xrange(N):
        lat, lon = map(float, fd.readline().split())
        coords.append((lat, lon))

for i in xrange(0, N):
    solution += haversine(coords[i][1], coords[i][0], coords[(i+1)%N][1], coords[(i+1)%N][0] )

with open(sys.argv[2], mode='w') as fd:
    fd.write(str(solution))
    fd.write('\n')